### PR TITLE
Add logging when detecting negative points

### PR DESCRIPTION
**Pull request by zachfang**
Add three variables:
detect_negative_points: the flag to detect negative points
negative_point_height_threshold: height threshold for the detection 
negative_point_number_threshold: number threshold for message logging

### CODE REVIEW

File: pcl_ros/include/sixriver/gpu/filters/impl/voxel_grid.hpp [hunk 1]
```diff
@@ -335,6 +353,12 @@ sixriver::VoxelGrid<PointT>::applyFilter (PointCloud &output)
             int idx = ijk0 * divb_mul_[0] + ijk1 * divb_mul_[1] + ijk2 * divb_mul_[2];
             index_vector.push_back (cloud_point_index_idx (static_cast<unsigned int> (idx), *it));
         }
+
+        // Log WARN message if negative points detected
+        if (detect_negative_points_ && (number_of_negative_points > negative_point_number_threshold_))
+        {
+            ROS_WARN_THROTTLE(3.0, "Detecting %u point(s) below height threshold", number_of_negative_points);
```
Comment (barabas6river) [inline pcl_ros/include/sixriver/gpu/filters/impl/voxel_grid.hpp]: If we're trying to find the max that won't false alarm, throttling will only show us the first high reading every 3 seconds, but we'll still not be sure about the true highest. How often do you see this log message in e.g. a movement loop? Could we get away with not throttling it?

File: pcl_ros/include/sixriver/gpu/filters/voxel_grid.h [hunk 2]
```diff
@@ -595,10 +664,10 @@ namespace sixriver
       inline void 
       setMinimumPointsNumberPerVoxel (unsigned int min_points_per_voxel) { min_points_per_voxel_ = min_points_per_voxel; }
 
-	  /** \brief Return the minimum number of points required for a voxel to be used.
+	    /** \brief Return the minimum number of points required for a voxel to be used.
```
Comment (barabas6river) [inline pcl_ros/include/sixriver/gpu/filters/voxel_grid.h:667]: Is this an indentation fix?

**Review by barabas6river — APPROVED** [review]
Looks good. If we don't see the log messages often, consider un-throttling them so we can see the true maximum number of points below the ground.

**Comment by barabas6river** [comment]
Since there are a lot of code paths with similar code, do we have a way of sanity-checking that they are all the same and/or won't get corrupted by another change? With this floor-detector, I'm not too worried, but in general, adding some automated tests would be helpful for preventing accidental code-rot.

File: pcl_ros/include/sixriver/gpu/filters/impl/voxel_grid.hpp (hunk 1 above)
Comment (zachfang) [reply pcl_ros/include/sixriver/gpu/filters/impl/voxel_grid.hpp]: Yeah, I am fine with removing the throttle. On the other hand I also don't want to spam the log because the code will be executed at 30 hz, so if the edge case gets hit, we will log a lot of messages.
This is the test I ran yesterday, if we set the number threshold to 0 and the robot is static, it will occasionally split out log with 16-30 points below -0.1m. For movement loop, ~150 points was the highest I saw yesterday with around 20 loops.
https://app.logdna.com/b2e069b3c4/logs/view?apps=&hosts=link&tags=rhqStaging&q=below%20height%20threshold

File: pcl_ros/include/sixriver/gpu/filters/voxel_grid.h (hunk 2 above)
Comment (zachfang) [reply pcl_ros/include/sixriver/gpu/filters/voxel_grid.h:667]: Yes, I found the indention was wrong (not consistent with other functions)

File: pcl_ros/include/sixriver/gpu/filters/impl/voxel_grid.hpp [hunk 3]
```diff
@@ -311,10 +318,16 @@ sixriver::VoxelGrid<PointT>::applyFilter (PointCloud &output)
             int idx = ijk0 * divb_mul_[0] + ijk1 * divb_mul_[1] + ijk2 * divb_mul_[2];
             index_vector.push_back (cloud_point_index_idx (static_cast<unsigned int> (idx), *it));
         }
+        if (detect_negative_points_ && (number_of_negative_points > negative_point_number_threshold_))
+        {
+            ROS_WARN_THROTTLE(3.0, "Detecting %u point(s) below height threshold", number_of_negative_points);
```
Comment (DrCodeWizzard) [inline pcl_ros/include/sixriver/gpu/filters/impl/voxel_grid.hpp]: is there a reason why we are throttling data here?

**Review by DrCodeWizzard — APPROVED** [review]

File: pcl_ros/include/sixriver/gpu/filters/impl/voxel_grid.hpp (hunk 3 above)
Comment (zachfang) [reply pcl_ros/include/sixriver/gpu/filters/impl/voxel_grid.hpp]: The reason was to avoid spamming log, but I can remove that though
